core/provisioning-java/src/main/java/org/apache/syncope/core/provisioning/java/data/GroupDataBinderImpl.java

```diff
@@ -207,16 +207,43 @@ public void create(final Group group, final GroupTO groupTO) {
             group.setName(groupPatch.getName().getValue());
         }
 
+        PropagationByResource<String> propByRes = new PropagationByResource<>();
         // owner
         if (groupPatch.getUserOwner() != null) {
-            group.setUserOwner(groupPatch.getUserOwner().getValue() == null
-                    ? null
-                    : userDAO.find(groupPatch.getUserOwner().getValue()));
+            if (StringUtils.isBlank(groupPatch.getUserOwner().getValue())) {
+                if (group.getUserOwner() != null) {
+                    group.setUserOwner(null);
+                    propByRes.addAll(ResourceOperation.UPDATE, groupDAO.findAllResourceKeys(group.getKey()));
+                }
+            } else {
+                User userOwner = userDAO.find(groupPatch.getUserOwner().getValue());
+                if (userOwner != null) {
+                    group.setUserOwner(userOwner);
+                    propByRes.addAll(ResourceOperation.UPDATE, groupDAO.findAllResourceKeys(group.getKey()));
+                } else {
+                    LOG.debug("Unable to find user owner for group {} by key {}",
+                            group.getKey(), groupPatch.getUserOwner().getValue() );
+                    group.setUserOwner(null);
+                }
+            }
         }
         if (groupPatch.getGroupOwner() != null) {
-            group.setGroupOwner(groupPatch.getGroupOwner().getValue() == null
-                    ? null
-                    : groupDAO.find(groupPatch.getGroupOwner().getValue()));
+            if (StringUtils.isBlank(groupPatch.getGroupOwner().getValue())) {
```

Comment:
Same observations here as in the former block

Reply:
agree, done